pkg/webhook/servicecatalog/serviceinstance/validation/handler.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"context"
+	"net/http"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/webhookutil"
+
+	admissionTypes "k8s.io/api/admission/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type Validator func(ctx context.Context, req admission.Request, si *sc.ServiceInstance, l *webhookutil.TracedLogger) error
+
+// AdmissionHandler handles ServiceInstance
+type AdmissionHandler struct {
+	decoder *admission.Decoder
+	client  client.Client
+
+	CreateValidators []Validator
+	UpdateValidators []Validator
+}
+
+var _ admission.Handler = &AdmissionHandler{}
+var _ admission.DecoderInjector = &AdmissionHandler{}
+var _ inject.Client = &AdmissionHandler{}
+
+func NewAdmissionHandler() {
```

Comment:
hm.. shouldn't be

```go
func NewAdmissionHandler() {
  return &AdmissionHandler{
	UpdateValidators: []Validator{h.DenyPlanChangeIfNotUpdatable}
  }
}
```

right now this func does not make sens

Reply:
@mszostok the code you wrote won't compile. You cannot use *h* variable - it is not initialized/defined

Reply:
we were discussing different scenarios - the logic implemented by DenyPlanChange method needs a client which is injected by the kubebuilder. Passing also client will make the method signature more and more complex.